eclipse-collections-api/src/main/java/org/eclipse/collections/api/map/MapIterable.java

```diff
@@ -284,4 +285,42 @@
     {
         return Spliterators.spliterator(this.iterator(), (long) this.size(), 0);
     }
+
+    /**
+     *
+     * Applies an aggregate function over the map grouping results into a map based on the specific key and value groupBy functions.
+     * Aggregate results are allowed to be immutable as they will be replaced in place in the map. A second function
+     * specifies the initial "zero" aggregate value to work with (i.e. Integer.valueOf(0)).
+     *
+     * <pre>
+     * MapIterable&lt;String, Interval&gt; map = Maps.mutable.with("oneToFive", Interval.fromTo(1, 5), "sixToNine", Interval.fromTo(6, 9));
+     *
+     * MapIterable&lt;String, Long&gt; result = map.aggregateBy(
+     *         eachKey -> {
+     *             return eachKey.equals("oneToFive")  ? "lessThanSix" : "greaterOrEqualsToSix";
+     *         },
+     *         each -> each.sumOfInt(Integer::intValue),
+     *         () -> 0L,
```

Comment:
this should be `Integer.valueOf(0)` Because everywhere else is Integer/int and here it is Long.